src/include/txn_inline.h

```diff
@@ -1065,9 +1074,11 @@ __wt_txn_read(
      */
     if (prepare_upd != NULL) {
         WT_ASSERT(session, F_ISSET(prepare_upd, WT_UPDATE_PREPARE_RESTORED_FROM_DS));
-        if (retry &&
-          (prepare_upd->txnid == WT_TXN_ABORTED ||
-            prepare_upd->prepare_state == WT_PREPARE_RESOLVED)) {
+        WT_ORDERED_READ(prepare_state, upd->prepare_state);
+        if (retry && pre_prepare_state != WT_PREPARE_ROLLBACK_INPROGRESS &&
```

Comment:
I don't understand the need for the `pre_prepare_state` check here. Can you explain it in a comment if there is a requirement for it?

Reply:
When we walk the update chain and we see WT_PREPARE_ROLLBACK_INPROGRESS, then the update has been restored to the update chain already. No need to retry in this case.

Reply:
Fix in 5b055458d9053fab47ff3a29d70a26b4996c66e1